test/test_um2netcdf.py

```diff
@@ -674,3 +675,46 @@ def test_fix_cell_methods_keep_weeks():
     assert mod.method == cm.method
     assert mod.coord_names == cm.coord_names
     assert mod.intervals[0] == "week"
+
+
+LEVEL_HEIGHTS = [20.0003377]  # NB: technically only need [0]
+
+LEVEL_COORDS = {um2nc.MODEL_LEVEL_NUM: iris.coords.DimCoord(range(1, 39)),
+                um2nc.LEVEL_HEIGHT: iris.coords.DimCoord(LEVEL_HEIGHTS),
+                um2nc.SIGMA: iris.coords.AuxCoord(np.array([0.99771646]))}
+
+
+@dataclass
+class FakeCubeCoords:
+
+    def coord(self, key):
+        return LEVEL_COORDS[key]
+
+
+def test_fix_level_coord_rho(z_sea_rho_data, z_sea_theta_data):
+    assert LEVEL_COORDS[um2nc.MODEL_LEVEL_NUM].var_name is None
+    assert LEVEL_COORDS[um2nc.LEVEL_HEIGHT].var_name is None
+    assert LEVEL_COORDS[um2nc.SIGMA].var_name is None
+
+    cube = FakeCubeCoords()
+    rho = np.ones(z_sea_theta_data.shape) * LEVEL_HEIGHTS[0]
+    um2nc.fix_level_coord(cube, rho, z_sea_theta_data)
+
+    assert LEVEL_COORDS[um2nc.MODEL_LEVEL_NUM].var_name == "model_rho_level_number"
+    assert LEVEL_COORDS[um2nc.LEVEL_HEIGHT].var_name == "rho_level_height"
+    assert LEVEL_COORDS[um2nc.SIGMA].var_name == "sigma_rho"
+
+
+def test_fix_level_coord_theta(z_sea_rho_data, z_sea_theta_data):
+    cube = FakeCubeCoords()
+    um2nc.fix_level_coord(cube, z_sea_rho_data, z_sea_theta_data)
+
+    assert LEVEL_COORDS[um2nc.MODEL_LEVEL_NUM].var_name == "model_theta_level_number"
+    assert LEVEL_COORDS[um2nc.LEVEL_HEIGHT].var_name == "theta_level_height"
+    assert LEVEL_COORDS[um2nc.SIGMA].var_name == "sigma_theta"
+
+
+def test_fix_level_coord_not_found(z_sea_rho_data, z_sea_theta_data):
```

Comment:
I'm wondering whether it would be worth having a quick docstring noting that this is checking `fix_level_coord` doesn't do anything if a cube doesn't have the vertical coordinates.

Reply:
Yep, that's a good idea as the test function name is too vague & there's no explanation. I'll aim for something where the name briefly states a requirement, then use a docstring for the detail.